app/views/govuk_publishing_components/components/docs/heading.yml

```diff
@@ -24,10 +24,18 @@ examples:
       text: 'Original consultation'
       heading_level: 3
   different_font_sizes:
-    description: Choose a different font size. Valid options are 24 (24px) and 19 (19px), with the component defaulting to 27px, appropriate for a H1. This option is not tied to the heading_level option in order to give flexibility.
+    description: |
+      Choose a different font size, defaults to 27px. Valid options are:
+
+      - 48 (48px)
+      - 36 (36px)
+      - 24 (24px)
+      - 19 (19px)
```

Comment:
I've always found it slightly unintuitive that the headings in the Component gem doesn't emulate the Design System's `xl`, `l` etc naming convention whilst emulating the heading styles. It also means that the style names lead to a dissonance between the name and what it does - for example, `.gem-c-heading--font-size-48` isn't always 48px in size.

Whilst the old naming convention shouldn't be removed for backwards compatibility reasons, is it worth updating this component to match the Design System naming convention? 

The number-based names could still be kept as an undocumented option for compatibility - but in this pull request the names could be updated to match the Design System's names. It's not a breaking change, since the old sizes are still useable.

Using the DS styles could mean less CSS shippped since the components could use `govuk-heading-*` classes, and we could remove some of the `gem-heading--*` classes.

Reply:
I share @injms's view that we should try and deprecate the value-based class names in favour of T-shirt sizes for both consistency with the Design System and, more importantly, maintainability.

Reply:
Thanks for the thinking. I've reworked the component to allow this, can you take another look?